src/components/ble/ImmediateAlertService.cpp

```diff
@@ -63,7 +63,8 @@ int ImmediateAlertService::OnAlertLevelChanged(uint16_t attributeHandle, ble_gat
       auto* alertString = ToString(alertLevel);
 
       NotificationManager::Notification notif;
-      std::memcpy(notif.message.data(), alertString, strlen(alertString));
+      std::memcpy(notif.message.data(), alertString, strlen(alertString) + 1);
```

Comment:
Is this change related to this patchset? Or is it an unrelated bug fix

Reply:
This is unrelated fix.

Reply:
I think it's to do with the fact that we don't use the notification buffer optimally (or even correctly really). I've proposed fixes for this in #1694 and #1695, but I haven't followed up on them.

Reply:
@NeroBurner Sure! I will make a separate PR for this. 
@FintasticMan I think when copying a string using `memcpy()` the size always needs to be `strlen()` of the string plus 1 for the trailing zero-byte string terminator (unless the destination memory is guaranteed to be prefilled with zero-bytes).

Reply:
@NeroBurner Created the PR #2159